### PR TITLE
Implement APIBlueprintWriter

### PR DESCRIPTION
**Pull request by cilim**
Hello API Hogs company 👋 

I love your library and I created a writer that supports API Blueprint markdown generation.

I've been using this library with my writer for some time now on a couple of production apps and I think it would be awesome if you could review this pull request and hopefully merge it to master so everyone can benefit from it. 😃 

Regards!

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -30,7 +30,8 @@ defmodule Bureaucrat.Mixfile do
      {:plug, "~> 1.0"},
      {:poison, "~> 1.5 or ~> 2.0 or ~> 3.0"},
      {:phoenix, "~> 1.2", optional: true},
-     {:ex_doc, ">= 0.0.0", only: :dev}
+     {:ex_doc, ">= 0.0.0", only: :dev},
+     {:inflex, "~> 1.10.0"}
```
Comment (mbuhot) [inline mix.exs]: Where is inflex used?

File: lib/bureaucrat/helpers.ex [hunk 2]
```diff
@@ -109,4 +109,12 @@ defmodule Bureaucrat.Helpers do
       group_title_for(mod, paths)
     end
   end
+
+  def plug_doc(conn, module: module, action: action) do
```
Comment (mbuhot) [inline lib/bureaucrat/helpers.ex]: A `@doc` string on this one will be helpful.

File: README.md [hunk 3]
```diff
@@ -200,4 +233,5 @@ be written to `web/controllers/api/v1/README.md`.
 * `:titles`: Allows you to specify explicit titles for some of your modules.
 For example `[{YourApp.Api.V1.UserController, "API /v1/users"}]` will
 change the title (Table of Contents entry and heading) for this controller.
+* `:prefix`: Allows you to remove the prefix of the test module names
```
Comment (mbuhot) [inline README.md]: Where is this config used?

Comment (cilim) [reply README.md]: Here: https://github.com/api-hogs/bureaucrat/pull/34/files#diff-d36f774419f37ac24b6ba39063685d10R144

File: mix.exs (hunk 1 above)
Comment (cilim) [reply mix.exs]: Here: https://github.com/api-hogs/bureaucrat/pull/34/files#diff-d36f774419f37ac24b6ba39063685d10R149

**Comment by cilim** [comment]
@mbuhot I accidentally pushed the wrong markdown file in the first commit. Thanks for reviewing the PR.

I just corrected the markdown writer and added the @doc to the `plug_doc()` function.

File: lib/bureaucrat/api_blueprint_writer.ex [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+  defp write_action(action, controller, records, file) do
+    test_description = "#{controller} #{action}"
+    record_request = Enum.at(records, 0)
+    method = record_request.method
+    path_params = record_request.path_params
+
+    file |> puts("### #{test_description} [#{method} #{anchor(record_request)}]")
+
+    unless path_params == %{} do
```
Comment (OpakAlex) [inline lib/bureaucrat/api_blueprint_writer.ex]: can we extract this to function, and maybe use pattern matching we can remove unless. Like,
```elixir
defp write_paramters(_path_params =%{}, file) do
  nil
end
defp write_parameters(path_params, file)
file
     |> puts("\n+ Parameters\n#{formatted_params(path_params)}")

      Enum.each(path_params, fn {param, value} ->
        puts(file, indent_lines(12, "#{param}: #{value}"))
      end)
end
```

File: lib/bureaucrat/api_blueprint_writer.ex [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      case record.query_string do
+        "" -> record.request_path
+        str -> "#{record.request_path}?#{str}"
+      end
+
+    file
+    |> puts("\n\n+ Request #{record.assigns.bureaucrat_desc}")
+    |> puts("**#{record.method}**&nbsp;&nbsp;`#{path}`\n")
+
+    unless record.req_headers == [] do
```
Comment (OpakAlex) [inline lib/bureaucrat/api_blueprint_writer.ex]: same refactor can be done here.

File: lib/bureaucrat/api_blueprint_writer.ex [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+      |> puts(indent_lines(4, "+ Headers\n"))
+
+      Enum.each(record.req_headers, fn {header, value} ->
+        puts(file, indent_lines(12, "#{header}: #{value}"))
+      end)
+
+      file
+    end
+
+    unless record.body_params == %{} do
```
Comment (OpakAlex) [inline lib/bureaucrat/api_blueprint_writer.ex]: same

File: lib/bureaucrat/api_blueprint_writer.ex [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+      file
+      |> puts(indent_lines(4, "+ Body\n"))
+      |> puts(indent_lines(12, format_body_params(record.body_params)))
+    end
+
+    file
+    |> puts("\n+ Response #{record.status}\n")
+
+    unless record.resp_headers == [] do
```
Comment (OpakAlex) [inline lib/bureaucrat/api_blueprint_writer.ex]: same

**Comment by cilim** [comment]
@OpakAlex Totally forgot about those Ruby leftovers xD
Thanks for the review :)

PR is updated!

**Comment by cilim** [comment]
Awesome! Let me know when you make a new release :)

**Pull request merged**